packages/clerk-js/src/ui/components/SessionTasks/tasks/TaskResetPassword/index.tsx

```diff
@@ -0,0 +1,188 @@
+import { useClerk, useReverification } from '@clerk/shared/react';
+import type { UserResource } from '@clerk/shared/types';
+import { useCallback } from 'react';
+
+import { useEnvironment, useSignOutContext, withCoreSessionSwitchGuard } from '@/ui/contexts';
+import { Col, descriptors, Flow, localizationKeys, useLocalizations } from '@/ui/customizables';
+import { Card } from '@/ui/elements/Card';
+import { useCardState, withCardStateProvider } from '@/ui/elements/contexts';
+import { Form } from '@/ui/elements/Form';
+import { Header } from '@/ui/elements/Header';
+import { useConfirmPassword } from '@/ui/hooks';
+import { useMultipleSessions } from '@/ui/hooks/useMultipleSessions';
+import { handleError } from '@/ui/utils/errorHandler';
+import { createPasswordError } from '@/ui/utils/passwordUtils';
+import { useFormControl } from '@/ui/utils/useFormControl';
+
+import { withTaskGuard } from './withTaskGuard';
+
+const TaskResetPasswordInternal = () => {
+  const { signOut, user, session } = useClerk();
+  const card = useCardState();
+  const {
+    userSettings: { passwordSettings },
+  } = useEnvironment();
+
+  const { t, locale } = useLocalizations();
+  const { otherSessions } = useMultipleSessions({ user });
+  const { navigateAfterSignOut, navigateAfterMultiSessionSingleSignOutUrl } = useSignOutContext();
+  const updatePasswordWithReverification = useReverification(
+    (user: UserResource, opts: Parameters<UserResource['updatePassword']>) => user.updatePassword(...opts),
+  );
+
+  const handleSignOut = () => {
+    if (otherSessions.length === 0) {
+      return signOut(navigateAfterSignOut);
+    }
+
+    return signOut(navigateAfterMultiSessionSingleSignOutUrl, { sessionId: session?.id });
+  };
+
+  const passwordField = useFormControl('newPassword', '', {
+    type: 'password',
+    label: localizationKeys('formFieldLabel__newPassword'),
+    isRequired: true,
+    validatePassword: true,
+    buildErrorMessage: errors => createPasswordError(errors, { t, locale, passwordSettings }),
+  });
+
+  const confirmField = useFormControl('confirmPassword', '', {
+    type: 'password',
+    label: localizationKeys('formFieldLabel__confirmPassword'),
+    isRequired: true,
+  });
+
+  const sessionsField = useFormControl('signOutOfOtherSessions', '', {
+    type: 'checkbox',
+    label: localizationKeys('formFieldLabel__signOutOfOtherSessions'),
+    defaultChecked: true,
+  });
+
+  const { setConfirmPasswordFeedback, isPasswordMatch } = useConfirmPassword({
+    passwordField,
+    confirmPasswordField: confirmField,
+  });
+
+  const canSubmit = isPasswordMatch;
+
+  const validateForm = () => {
+    if (passwordField.value) {
+      setConfirmPasswordFeedback(confirmField.value);
+    }
+  };
+
+  const resetPassword = useCallback(async () => {
+    if (!user) {
+      return;
+    }
```

Comment:
We could also add an assertion for `clerk.isLoaded` before declaring the functions / variable within this component, and then TypeScript would guard against not loaded resources, such as user.

The components only get rendered when `Clerk` is loaded anyway, so adding a comment to clarify that should suffice.